client/components/social-buttons/google.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import SocialLogo from 'social-logos';
+import { loadScript } from 'lib/load-script';
+
+export default class GoogleLoginButton extends Component {
+	static propTypes = {
+		clientId: PropTypes.string.isRequired,
+		scope: PropTypes.string,
+		fetchBasicProfile: PropTypes.bool,
+		responseHandler: PropTypes.func.isRequired,
+	};
+
+	static defaultProps = {
+		scope: 'profile',
+		fetchBasicProfile: true,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.googleAuth = null;
+		this.handleClick = this.handleClick.bind( this );
+	}
+
+	componentWillMount() {
+		this.loadGoogleAuth();
+	}
+
+	loadDependency() {
+		if ( window.gapi ) {
+			return Promise.resolve( window.gapi );
+		}
+
+		return new Promise( resolve => {
+			loadScript( 'https://apis.google.com/js/platform.js', () => resolve( window.gapi ) );
+		} );
+	}
+
+	loadGoogleAuth() {
+		return new Promise( resolve => {
+			if ( this.googleAuth ) {
```

Comment:
The problem with that approach:
```js
this.loadGoogleAuth().then( () => conosole.log( '1' ) );
this.loadGoogleAuth().then( () => conosole.log( '2' ) );
this.loadGoogleAuth().then( () => conosole.log( '3' ) );
```

This code will initiate 3 load requests, because we're setting `this.googleAuth` only after we finished loading,
we should "lock" when we start loading.